cnd/src/config/settings.rs

```diff
@@ -12,6 +12,7 @@ pub struct Settings {
     pub comit: Comit,
     pub network: Network,
     pub http_api: HttpSocket,
+    pub database: Option<Database>,
```

Comment:
The `Settings` struct is supposed to have all optional elements applied already, i.e. when converting from `File` to `Settings`, all the defaults are filled in. This also seems to be the case for the `database` config option (we always need a database).

Reply:
Oh, yes.  Will fix, thanks.

Reply:
This will probably bubble up to not accepting an `Option` in your constructor of the SqliteMetadataStore.

Reply:
> Why is web_gui optional in Settings?

Because it is actually optional at runtime.

Reply:
> Alright, a lot of round-a-bout comments here (from me). Eventually InMemoryDataStore will go away. I'm going to leave the config stuff as it is now and fix it then. There are non-trivial complications with config of the database and calling the db constructor iff both memory stores are in use - that is why this PR has the 'disable Sqlite' patch at the end. Admittedly its ugly.

Fair :)
Although the only thing the settings struct will contain is the default path right? What is the difference between setting the default path here if it does not exist vs. having an `Option` in the constructor?